### PR TITLE
Adds support for topology endpoints

### PR DESCRIPTION
**Pull request by zduey**
# Description

The python client did not have a way of hitting two endpoints:
- `/datasets/{datasetId}/concepts/{id}/topology`
- `/datasets/{datasetId}//concepts/schema/graph/`

This PR adds methods to hit these endpoints as well as some refactoring the testing setup to allow for independent test datasets to be created. This was necessary to be able to add tests for the new methods whose results would have not been as easily tested (and would change if future tests are added) if the session-level test dataset had been used.

The `/concepts/schema/graph/` endpoint in the concepts service that returns summary information about the Knowledge Graph that would be useful to access in the client. Currently, a list of dictionaries is returned where each dictionary represents either a `Model` or a `Relationship`. The added method determines which of the two it is, and unmarshals it into that object. What is returned is a dictionary containing two keys (models and relationships) whose values are a list of the un-marshalled objects of that type. 

The `/datasets/{datasetId}/concepts/{id}/topology` returns a list of models that have some defined relationship with the queried model.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Ran unit tests locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: blackfynn/api/concepts.py [hunk 1]
```diff
@@ -147,6 +147,41 @@ def files(self, dataset, concept, instance):
             ))
         return [DataPackage.from_dict(pkg, api=self.session) for r,pkg in resp]
 
+    def get_related(self, dataset, concept):
+        assert isinstance(concept, Model), "concept must be type Model"
```
Comment (merky) [inline blackfynn/api/concepts.py]: Looks familiar 🤔

File: blackfynn/models.py [hunk 2]
```diff
@@ -1913,6 +1913,14 @@ def remove_collaborators(self, *collaborator_ids):
         self._check_exists()
         return self._api.datasets.remove_collaborators(self, *collaborator_ids)
 
+    def get_topology(self):
+        """ ??
```
Comment (merky) [inline blackfynn/models.py]: Missing docstring.

File: blackfynn/models.py [hunk 3]
```diff
@@ -2931,6 +2939,18 @@ def delete_records(self, *records):
         for error in result['errors']:
             self._logger.error("Failed to delete instance {} with error: {}".format(error[0], error[1]))
 
+    def get_related(self):
+        """ Returns a list of related model types and counts of those relationships
```
Comment (merky) [inline blackfynn/models.py]: Looks familiar 🤔

File: tests/test_concepts.py [hunk 4]
```diff
@@ -333,3 +337,61 @@ def test_model_properties_with_enum(dataset):
...
     assert (array_property.multi_select == True)
     assert (array_property.enum == ['foo', 'bar', 'baz'])
+
+
+
+Graph = namedtuple('TestGraph', ['dataset', 'models', 'model_records',
+                                 'relationships', 'relationship_records'])
+
+
+@pytest.fixture(scope="module")
```
Comment (mattusifer) [inline tests/test_concepts.py]: 👍

**Pull request merged**